mongo/mongo.go

```diff
@@ -344,7 +348,10 @@ func (m *Mongo) UpsertVersion(id string, version *models.Version) (err error) {
 		},
 	}
 
-	_, err = s.DB(m.Database).C("versions").UpsertId(id, update)
+	info, err := s.DB(m.Database).C("versions").UpsertId(id, update)
+	if info.Updated == 0 {
```

Comment:
If no documents are updated we'll create all the dimensions? Not sure the creating of dimensions should live here really (happy to have a chat about it, I have no useful suggestions for alternatives)

Reply:
I added a comment on why I check if a document has been updated. Hope that helps